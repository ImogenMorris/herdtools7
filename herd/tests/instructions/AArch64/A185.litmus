AArch64 A185
{
int t[2] = {1,2};
0:X2=t;
1:X2=t;
}
  P0           |  P1              ;
LDR W0,[X2],#4 | ADD X2,X2,#8     ;  
LDR W1,[X2]    | LDR W0,[X2,#-4]! ;
               | LDR W1,[X2,#-4]! ;
forall
0:X0=1 /\ 0:X1=2
/\
1:X0=2 /\ 1:X1=1

