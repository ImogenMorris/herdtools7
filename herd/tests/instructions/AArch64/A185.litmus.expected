Test A185 Required
States 1
0:X0=1; 0:X1=2; 1:X0=2; 1:X1=1;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X0=1 /\ 0:X1=2 /\ 1:X0=2 /\ 1:X1=1)
Observation A185 Always 1 0
Hash=e9b676005c1f5a58e99d7a63485b8acc

